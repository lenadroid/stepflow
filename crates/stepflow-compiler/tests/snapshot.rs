use std::fs::File;
use std::io::BufReader;

use stepflow_compile::compile;
use stepflow_steps::Plugins;

#[test]
fn inputs_snapshot() {
    let plugins = Plugins::new();

    insta::glob!("inputs/*.yaml", |path| {
        let file = File::open(path).unwrap();
        let reader = BufReader::new(file);

        let flow = stepflow_workflow::Flow::from_yaml_reader(reader).unwrap();
        let result = compile(&plugins, flow).map_err(|e| e.to_string());

        insta::assert_yaml_snapshot!(result)
    })
}