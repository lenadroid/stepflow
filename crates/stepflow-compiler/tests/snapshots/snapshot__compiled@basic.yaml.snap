---
source: crates/stepflow-compiler/tests/snapshot.rs
expression: compiled
input_file: crates/stepflow-compiler/tests/valid/basic.yaml
---
input_schema:
  object:
    properties:
      in1:
        type: number
      in2:
        type: string
steps:
  - id: s1
    component: "mock://two_outputs"
    args:
      foo:
        input: in1
      bar:
        literal: 42
  - id: s2
    component: "mock://one_output"
    args:
      a:
        step: s1
        field: ~
      b:
        step: s1
        field: ~
      c:
        literal: 5
outputs:
  out1:
    step: s1
    field: ~
  out2:
    step: s1
    field: ~
  outLiteral:
    literal: 5
